cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServlet.java

```diff
@@ -655,62 +469,37 @@ private boolean getClusterLoad(HttpServletRequest request, HttpServletResponse r
 
   private boolean getPartitionLoad(HttpServletRequest request, HttpServletResponse response) throws Exception {
     Resource resource;
-    Long startMs;
-    Long endMs;
-    Integer entries;
-    Pattern topic;
+    long startMs;
+    long endMs;
+    Pattern topic = KafkaCruiseControlServletUtils.topic(request);
     int partitionUpperBoundary;
     int partitionLowerBoundary;
-    boolean json = wantJSON(request);
+    boolean json = KafkaCruiseControlServletUtils.wantJSON(request);
+    String resourceString = KafkaCruiseControlServletUtils.resourceString(request);
```

Comment:
The validation was intentionally kept out of `KafkaCruiseControlServletUtils.resourceString()`, because we would like to let user know about the potentially incorrect `resourceString` s/he provided to the servlet via passing `errorMsg` in `handleParameterParseException`.